cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -680,11 +700,21 @@ func instanceStatusFromProvisioningState(provisioningState *string) *cloudprovid
 	case string(compute.ProvisioningStateCreating):
 		status.State = cloudprovider.InstanceCreating
 	case string(compute.ProvisioningStateFailed):
-		status.State = cloudprovider.InstanceCreating
-		status.ErrorInfo = &cloudprovider.InstanceErrorInfo{
-			ErrorClass:   cloudprovider.OutOfResourcesErrorClass,
-			ErrorCode:    "provisioning-state-failed",
-			ErrorMessage: "Azure failed to provision a node for this node group",
+		// Provisioning can fail both during instance creation or after the instance is running.
+		// Per https://learn.microsoft.com/en-us/azure/virtual-machines/states-billing#provisioning-states,
+		// ProvisioningState represents the most recent provisioning state, therefore only report
+		// InstanceCreating errors when the power state indicates the instance has not yet started running
+		if !isRunningPowerState(powerState) {
+			klog.V(4).Infof("VM %s reports failed provisioning state with non-running power state: %s", resourceId, powerState)
+			status.State = cloudprovider.InstanceCreating
+			status.ErrorInfo = &cloudprovider.InstanceErrorInfo{
+				ErrorClass:   cloudprovider.OutOfResourcesErrorClass,
```

Comment:
I don't think we should assumed all errors to be `OutOfResourcesErrorClass`. The case where other errors like #5548 could happen again.

Reply:
I see your point—let's keep it that way.